### PR TITLE
Add missing docs/test to the Risc0 adapter

### PR DESCRIPTION
**Pull request by preston-evans98**
# Description
Add all missing docs to the `risc0` adapter. Add an integration test ensuring the correct behavior of the simulator.

## Linked Issues
- Related to #438

### CODE REVIEW

File: adapters/risc0/README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Risc0 Adapter
 
-This package adapts Risc0 version 0.14 to work as a ZKVM for the Sovereign SDK.
+This package adapts Risc0 version 0.18 to work as a ZKVM for the Sovereign SDK.
 
 ## Limitations
 
-Since recursion is not included in the 0.14 release, this adapter is currently limited - individual "slots" may
+Since in-VM recursion is not included in the 0.18 release, this adapter is currently limited. Individual "slots" may
```
Comment (orkunkilic) [inline adapters/risc0/README.md:7]: Risc0 calls this user recursion I believe. We can be consistent if you think it is important. FYI.

File: adapters/risc0/src/metrics.rs [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 use parking_lot::Mutex;
 use risc0_zkvm::Bytes;
 
+/// A global hashmap mapping metric names to their values.
 pub static GLOBAL_HASHMAP: Lazy<Mutex<HashMap<String, (u64, u64)>>> =
     Lazy::new(|| Mutex::new(HashMap::new()));
 
+/// Increments the requested metric by the given value, creating a
+/// new entry in the global map if necessary.
 pub fn add_value(metric: String, value: u64) {
```
Comment (citizen-stig) [inline adapters/risc0/src/metrics.rs]: This can be private to the module

File: adapters/risc0/src/metrics.rs [hunk 3]
```diff
@@ -19,6 +23,8 @@ pub fn add_value(metric: String, value: u64) {
         .or_insert((value, 1));
 }
 
+/// Deserialize a `Bytes` into a null-separated `(String, u64)` tuple. This function
+/// expects its arguments to match the format of arguments to Risc0's io callbacks.
 pub fn deserialize_custom(serialized: Bytes) -> Result<(String, u64), anyhow::Error> {
```
Comment (citizen-stig) [inline adapters/risc0/src/metrics.rs]: This can also be private

**Review by citizen-stig — APPROVED** [review]

File: adapters/risc0/README.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 # Risc0 Adapter
 
-This package adapts Risc0 version 0.14 to work as a ZKVM for the Sovereign SDK.
+This package adapts Risc0 version 0.18 to work as a ZKVM for the Sovereign SDK.
```
Comment (citizen-stig) [inline adapters/risc0/README.md]: Could we also stick to convention of spelling zkVM in #707 
```suggestion
This package adapts Risc0 version 0.18 to work as a zkVM for the Sovereign SDK.
```

File: adapters/risc0/src/metrics.rs (hunk 2 above)
Comment (bkolad) [reply adapters/risc0/src/metrics.rs]: fixed

File: adapters/risc0/README.md (hunk 1 above)
Comment (bkolad) [reply adapters/risc0/README.md:7]: fixed

**Pull request merged**